lib/secret-storage/tests/test_secret_storage.c

```diff
@@ -276,8 +296,8 @@ Test(secretstorage, test_rlimit)
 
   sprintf(key_fmt, "key%03d", i);
 
-  /* root is not restricted by rlimit */
-  if (geteuid() == 0)
+#if SYSLOG_NG_ENABLE_LINUX_CAPS
```

Comment:
As much as I understand these asserts test linux functionality, I think maybe it is time to let them go. As with this patch if you do not have or want libcap they are not existing anyway.
Otherwise if you still want to keep these asserts, please leave a space for my second patch not yet on master that forces the false branch execution. There is a usecase where custome memory management act like the false branch: https://github.com/Kokan/syslog-ng/commit/275b8eef3cdb1a27ff50d616ea057e74bf617b1e, https://github.com/Kokan/syslog-ng/commit/e352f30b4b4e60a7f677af7892e069b2c7cf40bd my patches for that.

Reply:
I agree that introducing these kind of dependencies in unit tests is not really fortunate and we will never notice if the test stops executing for any reason.

Can't we try if CAP_LOCK works simply by trying it in the test program? Or use mocking?

Reply:
I agree with both of you, I think it would be the best to remove this, if it causes this much trouble and complexity. @furiel ?

Reply:
Yes. Just remove it.